pkg/solana/client/multinode/client_test.go

```diff
@@ -0,0 +1,152 @@
+package client
+
+import (
+	"context"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/client/multinode/config"
+)
+
+type TestRPC struct {
+	latestBlock int64
+}
+
+type TestHead struct {
+	blockNumber int64
+}
+
+func (t *TestHead) BlockNumber() int64        { return t.blockNumber }
+func (t *TestHead) BlockDifficulty() *big.Int { return nil }
+func (t *TestHead) IsValid() bool             { return true }
+
+func LatestBlock(ctx context.Context, rpc *TestRPC) (*TestHead, error) {
+	rpc.latestBlock++
+	return &TestHead{rpc.latestBlock}, nil
+}
+
+func initializeMultiNodeClient(t *testing.T) *MultiNodeClient[TestRPC, *TestHead] {
+	requestTimeout := 5 * time.Second
+	lggr := logger.Test(t)
+	cfg := &config.MultiNodeConfig{}
+	cfg.SetDefaults()
+	enabled := true
+	cfg.MultiNode.Enabled = &enabled
+
+	c, err := NewMultiNodeClient[TestRPC, *TestHead](cfg, &TestRPC{}, requestTimeout, lggr, LatestBlock, LatestBlock)
+	require.NoError(t, err)
```

Comment:
Close client on clean-up, otherwise we can endup with flaky failure due to go routine leak